experimental/ast/nodes.go

```diff
@@ -43,6 +43,14 @@ func (n *Nodes) Root() File {
 	return File{wrapDeclBody(n.Context, 1)}
 }
 
+// NewPath creates a new Path with the given start and end nodes.
+//
+// Intended for use by the parser only.
+func (n *Nodes) NewPath(_ priv, start, end token.Token) Path {
```

Comment:
There will be a way to create synthetic paths, which will use the `token.synth` representation. Constructing natural paths except with the parser is not something we should allow at all, because `Path` makes fairly strong assumptions about the token stream inside of it. I have been punting on designing this API.

I think this function should be renamed to `NewPathInternal`, though.

Reply:
Removed the function.